samza-core/src/main/java/org/apache/samza/system/inmemory/InMemorySystemAdmin.java

```diff
@@ -0,0 +1,154 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.system.inmemory;
+
+import com.google.common.base.Strings;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.samza.config.Config;
+import org.apache.samza.system.StreamSpec;
+import org.apache.samza.system.StreamValidationException;
+import org.apache.samza.system.SystemAdmin;
+import org.apache.samza.system.SystemStreamMetadata;
+import org.apache.samza.system.SystemStreamPartition;
+
+
+/**
+ * Initial draft of in-memory {@link SystemAdmin}. It is test only and not meant for production use right now.
+ */
+public class InMemorySystemAdmin implements SystemAdmin {
+  private final InMemoryManager inMemoryManager;
+  private final InMemorySystemConfig inMemorySystemConfig;
+
+  public InMemorySystemAdmin(InMemoryManager manager, Config config) {
+    inMemoryManager = manager;
+    inMemorySystemConfig = new InMemorySystemConfig(config);
+  }
+
+  @Override
+  public void start() {
+
+  }
+
+  @Override
+  public void stop() {
+
+  }
+
+  /**
+   * Fetches the offsets for the messages immediately after the supplied offsets
+   * for a group of SystemStreamPartitions.
+   *
+   * @param offsets
+   *          Map from SystemStreamPartition to current offsets.
+   * @return Map from SystemStreamPartition to offsets immediately after the
+   *         current offsets.
+   */
+  @Override
+  public Map<SystemStreamPartition, String> getOffsetsAfter(Map<SystemStreamPartition, String> offsets) {
+    return offsets.entrySet()
+        .stream()
+        .collect(Collectors.toMap(Map.Entry::getKey, entry -> {
+            String offset = entry.getValue();
+            return String.valueOf(Integer.valueOf(offset) + 1);
+          }));
+  }
+
+  /**
+   * Fetch metadata from a system for a set of streams.
+   *
+   * @param streamNames
+   *          The streams to to fetch metadata for.
+   * @return A map from stream name to SystemStreamMetadata for each stream
+   *         requested in the parameter set.
+   */
+  @Override
+  public Map<String, SystemStreamMetadata> getSystemStreamMetadata(Set<String> streamNames) {
+    return inMemoryManager.getSystemStreamMetadata(streamNames);
+  }
+
+  /**
+   * Compare the two offsets. -1, 0, +1 means offset1 &lt; offset2,
+   * offset1 == offset2 and offset1 &gt; offset2 respectively. Return
+   * null if those two offsets are not comparable
+   *
+   * @param offset1 First offset for comparison.
+   * @param offset2 Second offset for comparison.
+   * @return -1 if offset1 &lt; offset2; 0 if offset1 == offset2; 1 if offset1 &gt; offset2. Null if not comparable
+   */
+  @Override
+  public Integer offsetComparator(String offset1, String offset2) {
+    int o1 = Optional.ofNullable(offset1)
```

Comment:
this logic is too hard to understand. I would refer a simple something like:

if(offset1 == null || offset2 == null) {
   return null;
} else {
   return Integer.valueOf(offset1).compare(Interger.valueOf(offset2));
}